sklearn/mixture/bayesian_mixture.py

```diff
@@ -0,0 +1,980 @@
+        log_prob = -.5 * (
+            -log_lambda +
+            (n_features / self.mu_beta_ +
+             self.lambda_nu_ / self.lambda_inv_W_ *
+             (np.sum((self.mu_m_ ** 2), 1) -
+              2 * np.dot(X, self.mu_m_.T) +
+              np.sum(X ** 2, 1)[:, np.newaxis])))
+        log_prob -= .5 * n_features * np.log(2 * np.pi)
+        self._log_lambda = log_lambda
+        return log_prob
+
+    def _check_is_fitted(self):
+        check_is_fitted(self, 'weight_alpha_')
+        check_is_fitted(self, 'mu_beta_')
+        check_is_fitted(self, 'mu_m_')
+        check_is_fitted(self, 'lambda_nu_')
+        check_is_fitted(self, 'lambda_inv_W_')
+
+    def _get_parameters(self):
+        return (self.weight_alpha_, self.mu_beta_, self.mu_m_, self.lambda_nu_,
+                self.lambda_inv_W_)
+
+    def _set_parameters(self, params):
+        (self.weight_alpha_, self.mu_beta_, self.mu_m_, self.lambda_nu_,
+         self.lambda_inv_W_) = params
+
+    def _estimate_lower_bound(self, log_prob, resp, log_resp):
+        log_p_XZ = self._estimate_p_XZ(log_prob, resp)
+        log_p_weight = self._estimate_p_weight()
+        log_p_mu_lambda = self._estimate_p_mu_lambda()
+        log_q_z = self._estimate_q_Z(resp, log_resp)
+        log_q_weight = self._estimate_q_weight()
+        log_q_mu_lambda = self._estimate_q_mu_lambda()
+        # print log_p_XZ, log_p_weight, log_p_mu_lambda, log_q_z, \
+        #     log_q_weight, log_q_mu_lambda
+        return log_p_XZ + log_p_weight + log_p_mu_lambda + log_q_z + \
+            log_q_weight + log_q_mu_lambda
+
+    def _estimate_p_XZ(self, log_prob, resp):
+        """Equation 7.5, 7.6."""
+        return np.sum(log_prob * resp)
+
+    def _estimate_p_weight(self):
+        """Equation 7.7."""
+        return (self._log_dirichlet_norm_alpha_prior +
+                (self.weight_alpha_prior_ - 1) * np.sum(self._log_pi))
+
+    def _estimate_p_mu_lambda(self):
+        if self.precision_type == 'full':
+            return self._estimate_p_mu_lambda_full()
+        elif self.precision_type == 'tied':
+            return self._estimate_p_mu_lambda_tied()
+        elif self.precision_type == 'diag':
+            return self._estimate_p_mu_lambda_diag()
+        elif self.precision_type == 'spherical':
+            return self._estimate_p_mu_lambda_spherical()
+
+    def _estimate_p_mu_lambda_full(self):
+        """Equation 7.9."""
+        temp1 = (self._log_lambda -
+                 self.n_features * self.mu_beta_prior_ / self.mu_beta_)
+        mk_sol = np.empty(self.n_components)
+        for k in range(self.n_components):
+            sol = linalg.solve_triangular(
+                self._inv_W_chol[k],
+                (self.mu_m_[k] - self.mu_m_prior_).T, lower=True).T
+            mk_sol[k] = np.sum(np.square(sol))
+        temp2 = self.mu_beta_prior_ * self.lambda_nu_ * mk_sol
+        temp_mu = (.5 * np.sum(temp1 - temp2) +
+                   self._log_gaussian_norm_prior)
```

Comment:
My previous comment wasn't clear enough. I give more details here to be sure, we are ok.
I thought :
```
temp_mu = (.5 * np.sum(temp1 - temp2) +
                        self.n_components * self._log_gaussian_norm_prior)
```
According to Line532, `self._log_gaussian_norm_prior = D ln (beta_0 / (2pi))`
According to the formula 7.9 of your report, you do : `sum_{k=1^K}  (D ln ( beta/(2 pi))` 
Normaly on Line789, it should be : `self.n_components * self._log_gaussian_norm_prior`
Am I right ?